migration/workspace-query-separation/src/test/java/ddf/migration/workspace/query/separation/WorkspaceQueryMigrationTest.java

```diff
@@ -0,0 +1,109 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.migration.workspace.query.separation;
+
+import static org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceConstants.WORKSPACE_QUERIES;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import com.google.common.collect.ImmutableList;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.transform.InputTransformer;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.Collections;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class WorkspaceQueryMigrationTest {
+
+  @Mock private QueryResponse queryResponse;
+
+  @Mock private CatalogFramework catalogFramework;
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private FilterBuilder filterBuilder;
+
+  @Mock private InputTransformer xmlInputTransformer;
+
+  private static WorkspaceQueryMigration workspaceQueryMigration;
+
+  @Before
+  public void setup() throws Exception {
+    initMocks(this);
+
+    workspaceQueryMigration =
+        new WorkspaceQueryMigration(catalogFramework, filterBuilder, xmlInputTransformer);
+  }
+
+  @Test
+  public void testDataMigration() throws Exception {
+    Result result = mock(Result.class);
+    doReturn(getWorkspaceMetacard("workspaceId")).when(result).getMetacard();
+
+    doReturn(Collections.singletonList(result)).when(queryResponse).getResults();
+    doReturn(queryResponse).when(catalogFramework).query(any(QueryRequest.class));
+
+    doReturn(new MetacardImpl()).when(xmlInputTransformer).transform(any(InputStream.class));
+
+    Metacard metacard = new MetacardImpl();
+    metacard.setAttribute(new AttributeImpl(Core.ID, "queryId"));
+
+    CreateResponse createResponse = mock(CreateResponse.class);
+    doReturn(ImmutableList.of(metacard, metacard)).when(createResponse).getCreatedMetacards();
+
+    doReturn(createResponse).when(catalogFramework).create(any(CreateRequest.class));
+
+    workspaceQueryMigration.migrateWorkspaces();
+
+    verify(catalogFramework, times(1)).create(any(CreateRequest.class));
```

Comment:
We should capture the requests to create and update and perform assertions on them to ensure we are creating the right query metacards and updating the workspace metacard correctly.